File: Lesson1/Dice/Dice/LoadedDie.cs
using System;

namespace Dice
{
    internal class LoadedDie : IDie
    {
        private readonly int _sides;
        private readonly int _load;

        public LoadedDie(int sides, int load)
        {
            _sides = sides;
            _load = load;
        }

        public int Roll()
        {
            var rnd = RandomFeeder.Feeder;
            int roll;
            if (rnd.Next(1, 3) >= 2)
            {
                roll = this._load;
            }
            else
            {
                do
                {
                    roll = rnd.Next(1, _sides + 1);
                } while (roll == this._load);
            }

            return roll;
        }
    }
}

File: Lesson1/src/main/java/com/mikeabney/katas/dice/domain/LoadedDie.java
package com.mikeabney.katas.dice.domain;

import java.util.Random;

public class LoadedDie extends Die {
    private Random loadGenerator;
    private int loadedValue;

    public LoadedDie(int sides, int loadedValue) {
        super(sides);
        if (loadedValue < 1 || loadedValue > sides) {
            throw new IllegalArgumentException(
                    String.format("Attempt to load %i-sided die with value (%i) that doesn't match a side!",
                    sides, loadedValue));
        }
        loadGenerator = new Random(System.currentTimeMillis());
        this.loadedValue = loadedValue;
    }

    @Override
    public int roll() {
        if (loadGenerator.nextBoolean()) {
            return loadedValue;
        }
        return super.roll();
    }
}
